block/src/Pos/Block/Logic/VAR.hs

```diff
@@ -207,11 +219,11 @@ verifyAndApplyBlocks pm rollback blocks = runExceptT $ do
                        <> pretty epochIndex
             lift $ lrcSingleShot pm epochIndex
         logDebug "Rolling: verifying"
-        lift (verifyBlocksPrefix pm prefix) >>= \case
+        lift (verifyBlocksPrefix pm ctx prefix) >>= \case
             Left (ApplyBlocksVerifyFailure -> failure)
                 | rollback  -> failWithRollback failure blunds
                 | otherwise -> do
-                      logDebug "Rolling: Applying AMAP"
+                      logDebug $ sformat ("Rolling: Applying AMAP: "%shown) failure
```

Comment:
Consider improving this message even more? What is AMAP?

Reply:
it's apply as much as possible:  `applyAMAP` applies blocks one at a time, and it runs only if `verifyBlocksPrefix` failed on the supplied list of blocks.